src/server/game/Battlegrounds/ArenaTeamMgr.cpp

```diff
@@ -186,8 +186,9 @@ void ArenaTeamMgr::DistributeArenaPoints()
     {
         if (ArenaTeam* at = titr->second)
         {
-            at->FinishWeek();
-            at->SaveToDB();
+            if (at->FinishWeek())
+                at->SaveToDB();
+            
             at->NotifyStatsChanged();
```

Comment:
What I meant to say is, it could be done as following:
`if (!at->FinishWeek())
     continue;
at->SaveToDB();
at->NotifyStatsChanged();`

Because we don't really have to notify the members of their stats as there was no alterations made from their side as they didn't play this week but I rather chose to inform them as that's not the underlying problem that freezes the thread.

My commit should be accepted, imo.